src/etc/inc/openvpn.inc

```diff
@@ -1575,21 +1575,33 @@ function openvpn_resync_all($interface = "") {
 		log_error(gettext("Resyncing OpenVPN instances."));
 	}
 
+	// Check if OpenVPN clients and servers require a resync.
 	if (is_array($config['openvpn']['openvpn-server'])) {
 		foreach ($config['openvpn']['openvpn-server'] as & $settings) {
-			if ($interface <> "" && $interface != $settings['interface']) {
-				continue;
+			$mode_id = "server" . $settings['vpnid'];
+			$fpath = "{$g['varetc_path']}/openvpn/{$mode_id}.interface";
+			$gw_group_change = FALSE;
+			$ip_change = ($interface === $settings['interface']);
+			if (file_exists($fpath)) {
```

Comment:
if `$ip_change` is true the check for `$gw_group_change` could be skipped

Reply:
Noted, I will improve the efficiency here.

Reply:
This has been improved in changeset 15f8062b42b3b2849d5dd7fdde9170d4785e84e4